src/agent/src/sandbox.rs

```diff
@@ -388,16 +388,16 @@ fn online_resources(logger: &Logger, path: &str, pattern: &str, num: i32) -> Res
             }
             let c = c.unwrap();
 
-            if c.trim().contains('0') {
```

Comment:
Consider this case:
1. start  container A with 2 CPUs first
2. then start container B with 2 CPUs

After container A started, there may already 3 CPUs are online, when starting container B, assume that the CPU files are all filled by `1`, and they will be counted up, so there will no new CPU files will be updated from `0` to `1` for container B.

Does this scenario make an issue?

Reply:
@liubin I think that the whole algorithm is presently broken. The reason is that what we get from the runtime is `cpuDelta`, and I don't see any place where we ever add the number of expected CPUs. In other words, what we have is the number of CPUs that should be _added_, not the _total_ number of CPUs we expect.

Unfortunately, that does not work well with the curent approach where the code never counts the number of CPUs before adding them.

Reply:
@snir911 I believe that @liubin's comment is the most likely cause for the bug earlier. So I would suggest changing the commit description:

If a cpu was onlined earlier, for example because we ran another container that already onlined a few CPUs, the agent fails to use it. With this fix, we now ensure we have the total number of requested CPUs.

That being said, I believe that the new algorithm still fails in this scenario.

Reply:
When you start container B 2 additional CPUs are hot-plugged? it shouldn't be a problem as they weren't counted before.
Is there a scenario when CPU is plugged but stays offline? (IIUC it shouldn't be plugged if it won't be onlined)